client/state-db/src/lib.rs

```diff
@@ -59,10 +60,26 @@ use std::{
 };
 
 const PRUNING_MODE: &[u8] = b"mode";
+const PRUNING_MODE_ARG: &[u8] = b"mode_arg";
+
 const PRUNING_MODE_ARCHIVE: &[u8] = b"archive";
 const PRUNING_MODE_ARCHIVE_CANON: &[u8] = b"archive_canonical";
 const PRUNING_MODE_CONSTRAINED: &[u8] = b"constrained";
 
+/// This structure to serialize the values of pruning constraints.
+/// The [`Constraints`] are not used because scale-codec does not work with `usize` fields.
+#[derive(codec::Encode, codec::Decode)]
+struct LastUsedPruningModeConstraint {
+	max_blocks: Option<u32>,
+	max_mem: Option<u64>,
+}
+impl Default for LastUsedPruningModeConstraint {
```

Comment:
```suggestion

impl Default for LastUsedPruningModeConstraint {
```

nitpick, feel free to ignore